src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala

```diff
@@ -16,25 +16,51 @@ import scala.collection.JavaConverters._
 object Neo4jDataFrame {
 
   def mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (String,Seq[String]), relationship: (String,Seq[String]), target: (String,Seq[String])): Unit = {
+    mergeEdgeList(sc, dataFrame, source,relationship, target, Map.empty)
+  }
+
+  def mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (String,Seq[String]), relationship: (String,Seq[String]), target: (String,Seq[String]), renameMap: Map[String,String]): Unit = {
+    val sourceLbel = renameMap.getOrElse(source._2.head, source._2.head)
+    val targetLabel = renameMap.getOrElse(target._2.head, target._2.head)
     val mergeStatement = s"""
       UNWIND {rows} as row
-      MERGE (source:`${source._1}` {`${source._2.head}` : row.source.`${source._2.head}`}) ON CREATE SET source += row.source
-      MERGE (target:`${target._1}` {`${target._2.head}` : row.target.`${target._2.head}`}) ON CREATE SET target += row.target
+      MERGE (source:`${source._1}` {`${source._2.head}` : row.source.`${sourceLbel}`}) ON CREATE SET source += row.source
+      MERGE (target:`${target._1}` {`${target._2.head}` : row.target.`${targetLabel}`}) ON CREATE SET target += row.target
       MERGE (source)-[rel:`${relationship._1}`]->(target) ON CREATE SET rel += row.relationship
       """
     val partitions = Math.max(1,(dataFrame.count() / 10000).asInstanceOf[Int])
     val config = Neo4jConfig(sc.getConf)
     dataFrame.repartition(partitions).foreachPartition( rows => {
       val params: AnyRef = rows.map(r =>
         Map(
-          "source" -> source._2.map( c => (c, r.getAs[AnyRef](c))).toMap.asJava,
-          "target" -> target._2.map( c => (c, r.getAs[AnyRef](c))).toMap.asJava,
+          "source" -> source._2.map( c => (renameMap.getOrElse(c,c), r.getAs[AnyRef](c))).toMap.asJava,
+          "target" -> target._2.map( c => (renameMap.getOrElse(c,c), r.getAs[AnyRef](c))).toMap.asJava,
           "relationship" -> relationship._2.map( c => (c, r.getAs[AnyRef](c))).toMap.asJava)
           .asJava).asJava
           execute(config, mergeStatement, Map("rows" -> params).asJava, write = true)
     })
   }
 
+
+  def createNodes(sc: SparkContext, dataFrame: DataFrame, nodes: (String,Seq[String])): Unit = {
+
+    val nodeLabel = nodes._2.head
+    val createStatement = s"""
```

Comment:
no optional renaming here?

Reply:
With nodes creation it's different I guess.
2 columns with the same name aren't possible.

Reply:
I thought it as an additional aliasing.

Reply:
So it would be cool to add.